common/sgx/tlsverifier.c

```diff
@@ -172,3 +198,131 @@ oe_result_t oe_verify_attestation_certificate(
     oe_free(report);
     return result;
 }
+
+/**
+ * oe_verify_attestation_certificate_with_evidence
+ *
+ * This function performs a custom validation on the input certificate. This
+ * validation includes extracting attestation evidence extension from the
+ * certificate before validating this evidence. An optional
+ * claim_verify_callback could be passed in for a calling client to further
+ * validate the claims of the enclave creating the certificate.
+ * @param[in] cert_in_der a pointer to buffer holding certificate contents
+ *  in DER format
+ * @param[in] cert_in_der_len size of certificate buffer above
+ * @param[in] claim_verify_callback callback routine for custom claim checking
+ * @param[in] arg an optional context pointer argument specified by the caller
+ * when setting callback
+ * @retval OE_OK on a successful validation
+ * @retval OE_VERIFY_FAILED on quote failure
+ * @retval OE_INVALID_PARAMETER At least one parameter is invalid
+ * @retval OE_FAILURE general failure
+ * @retval other appropriate error code
+ */
+oe_result_t oe_verify_attestation_certificate_with_evidence(
+    uint8_t* cert_in_der,
+    size_t cert_in_der_len,
+    oe_verify_claims_callback_t claim_verify_callback,
+    void* arg)
+{
+    oe_result_t result = OE_FAILURE;
+    oe_cert_t cert = {0};
```

Comment:
https://github.com/openenclave/openenclave/blob/master/docs/Community/DevelopmentGuide.md says:

> DON'T use abbreviations unless they are already well-known terms known by users (e.g., "app", "info"), or are already required for use by developers (e.g, "min", "max", "args"). Examples of bad use would be num_widgets instead of widget_count, and opt_widgets instead of option_widgets or optional_widgets.

So: cert -> certificate, pub -> public, buf -> buffer

Reply:
IMHO, this guideline is arbitrary because each person may have difference list of `well-known terms`. In addition, the other functions in OE use the above mentioned abbreviations extensively.

Reply:
Yes the only terms that should be considered well known are ones explicitly mentioned as exceptions in the coding style guidelines (by default, nothing).   I agree that other functions in OE are bad and should be cleaned up too, but not in this PR.